test/unit/extract-query-params.test.js

```diff
@@ -0,0 +1,7 @@
+import extractQueryParams from "extractQueryParams";
+
+test("query param extraction", () => {
+  const [src, params] = extractQueryParams("https://demo.imgix.net/abc.png?w=100&auto=format,enhance")
+  expect(src).toEqual("https://demo.imgix.net/abc.png");
+  expect(params).toEqual({w: "100", auto: "format,enhance" });
+});
```

Comment:
I wouldn't mind seeing a test or two in here to check/enforce the decoding behavior that's also included in this function.

Reply:
Done!